### PR TITLE
Process footnotes.xml

### PR DESCRIPTION
**Pull request by kylegibson**
Reference: ECMA 376 17.11

A basic initial implementation for footnotes.

* This doesn't process `endnotes.xml`
* `numFmt` is ignored and always uses decimals
* Footnotes are always placed at the very end of the document, regardless of `pos`. HTML has no concept of pages, but it could handle the beneath text option.
* `numRestart` and `numStart` are ignored, so the footnotes always start at 1, and never restarts

### CODE REVIEW

File: pydocx/tests/test_html.py [hunk 1]
```diff
@@ -4,15 +4,268 @@
...
+        expected_html = '''
+            <p>
+                Foo
+                <sup>
+                    <a href="#footnote-abc" name="footnote-ref-abc">1</a>
+                </sup>
+            </p>
+            <p>Footnotes should appear below this</p>
+            <hr/>
+            <ol type="1">
```
Comment (jlward) [inline pydocx/tests/test_html.py]: Are we sure we want to hard code this to `type="1"`? Shouldn't we use `pydocx-decimal` or whatever the class is for decimal lists?

I don't think the semantinator will convert this correctly.

Comment (kylegibson) [inline pydocx/tests/test_html.py]: I did this because I thought it's what you had suggested. I can change it to whatever makes the most sense.

Comment (jlward) [inline pydocx/tests/test_html.py]: Ah yes, I see the comment now. I was not clear. We actually use `list-style-type` instead of `type`.

**Comment by jlward** [comment]
After you update from `ol type` to `ol class` this looks good.

**Pull request merged**